src/prefect/flow_engine.py

```diff
@@ -382,45 +381,27 @@ def get_hooks(self, state: State, as_async: bool = False) -> Iterable[Callable]:
             hooks = flow.on_crashed_hooks
         elif state.is_running() and flow.on_running_hooks:
             hooks = flow.on_running_hooks
+        else:
+            hooks = None
 
         for hook in hooks or []:
```

Comment:
```suggestion
        else:
            hooks = []

        for hook in hooks:
```

Maybe like 0.3% cleaner?

Reply:
every % counts